assets/src/components/ladderPage.tsx

```diff
@@ -143,6 +145,9 @@ const LadderPage = (): ReactElement<HTMLDivElement> => {
 
   const mobileMenuClass = mobileMenuIsOpen ? "blurred-mobile" : ""
 
+  const [showDetourModal, setShowDetourModal] = useState(false)
+  const [routeForDetour, setRouteForDetour] = useState<Route | null>(null)
```

Comment:
question: should it be possible to have the route closed while keeping the last route saved?

That doesn't seem like intended behavior to me, so I think we could derive `showDetourModal`'s state from `routeForDetour !== null` instead of keeping track of two states which are implicitly related.

Reply:
I was thinking the same thing. I took the general concept from `<MapPage />`, which does the same thing, but it does ... seem odd to me.

Reply:
I figured we did it in `<MapPage />` for a reason, but now that I think about it, it might have been more than I forgot to remove the unnecessary boolean in `<MapPage />` also 🤔

Reply:
[Fixed](https://github.com/mbta/skate/pull/2647/commits/dcdf89dfff324438f08a006e4c1bdb5584623896)

Reply:
Ooooh I know the reason....

It's because of the exit animation. If we derive this state from `routeForDetour !== null`, then the whole detour modal disappears from the DOM immediately, while having `showDetourModal` as a separate prop allows it to have an exit animation.